### ISSUE TITLE
Android 11 (R) Camera not found

### ISSUE THREAD

**Issue by Voonder** [esafirm/android-image-picker#296]
## Expected Behavior

The application find a camera to took a picture

## Actual Behavior

The application doesn't find any camera application.

## Steps to Reproduce the Problem

  1. Change compileSdk and targetSdk to 30
  1. Launch the demo app 
  1. Clic on the button "Camera"
  1.  Toast with message "No camera found"

## Specifications

  - Image Picker Version: 2.4.0
  - Android OS: Android 11 (R)
  - Phone: Pixel 3

--------------

When I looking on the official documentation, it's link to the new privacy update introduce in Android 11 ([Package Visibility](https://developer.android.com/about/versions/11/privacy/package-visibility))

You need to allow the Camera Intent to work with this special query in the Manifest with:

```
<queries>
    <intent>
        <action android:name="android.media.action.IMAGE_CAPTURE" />
    </intent>
</queries>
```
(found here : [https://stackoverflow.com/a/63246789/4101898](https://stackoverflow.com/a/63246789/4101898))

I try to implement it.
First, You need to download Android Studio 4.1 to have at least Android Gradle Plugin 4.1.0 (release candidate)
Next, Add the code above in the Manifest.

When I start the app and I try again. I have a crash in the library code:

```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.esafirm.imagepicker.sample, PID: 23850
    java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=553, result=-1, data=Intent { (has extras) }} to activity {com.esafirm.imagepicker.sample/com.esafirm.sample.MainActivity}: java.lang.NullPointerException: Attempt to invoke interface method 'java.lang.Object[] java.util.Collection.toArray()' on a null object reference
        at android.app.ActivityThread.deliverResults(ActivityThread.java:5015)
        at android.app.ActivityThread.handleSendResult(ActivityThread.java:5056)
        at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:51)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7656)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
     Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'java.lang.Object[] java.util.Collection.toArray()' on a null object reference
        at java.util.ArrayList.addAll(ArrayList.java:588)
        at com.esafirm.sample.MainActivity.onActivityResult(MainActivity.kt:108)
        at android.app.Activity.dispatchActivityResult(Activity.java:8310)
        at android.app.ActivityThread.deliverResults(ActivityThread.java:5008)
        at android.app.ActivityThread.handleSendResult(ActivityThread.java:5056) 
        at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:51) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:223) 
        at android.app.ActivityThread.main(ActivityThread.java:7656) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947) 
```

**Comment by esafirm** [comment]
@Voonder thanks for the report, will look into it

**Comment by simon-the-canadian** [comment]
So I updated my app to target API 30 and ran into this same issue. Only happens on Pixel phones running Android 11, though. My Samsung phone on Android 11 doesn't have this issue 🤷🏽‍♂️

**Comment by JamzWork** [comment]
Same Issue as Voonder:

```
java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=66089, result=-1, data=Intent { (has extras) }} to activity {com.app.TestApp}: java.lang.NullPointerException: images must not be null
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4789)
        at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4832)
        at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:52)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:190)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:105)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2386)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:213)
        at android.app.ActivityThread.main(ActivityThread.java:8178)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1101) 
```

**Comment by JenyaKirmiza** [comment]
any news? soon we must all change compileSdk and targetSdk to 30, any change it's gonna be fixed ? thanks anyway for the good lib

**Comment by audiserg** [comment]
<queries>
    <intent>
        <action android:name="android.media.action.IMAGE_CAPTURE" />
    </intent>
</queries>

Adding this to manifest(not in apllication or activity tag in manifest ) and in emulator api30  work fine

**Comment by esafirm** [comment]
This is already in main and will be included in 3.0.0 release.
